From 8d62991bb9e0bcb41a8fc2270664a6094a1909fe Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Tue, 4 Jul 2017 22:40:33 +0300
Subject: [PATCH 1772/2241] fix: dts: a70x0: a80x0: align DTS name of
 Armad-7/8k with mainline

update the device tree file name to be aligned with mainline,
remove the name "A" from file name.
no functional change, and update the documentation according to this change

Change-Id: I79b85a46b6bd372f62848d9e44c135846b3b4318
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/41153
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
---
 .../bindings/arm/marvell/armada-7k-8k.txt          |   4 +-
 .../mvebu/a7k-a8k/armada7040-db-setup.txt          |  14 +-
 .../mvebu/a7k-a8k/armada8040-db-setup.txt          |  20 +-
 arch/arm64/boot/dts/marvell/Makefile               |   4 +-
 arch/arm64/boot/dts/marvell/armada-7040-db-A.dts   | 177 ---------------
 arch/arm64/boot/dts/marvell/armada-7040-db.dts     | 177 +++++++++++++++
 arch/arm64/boot/dts/marvell/armada-8040-db-A.dts   | 238 ---------------------
 arch/arm64/boot/dts/marvell/armada-8040-db.dts     | 238 +++++++++++++++++++++
 8 files changed, 436 insertions(+), 436 deletions(-)
 delete mode 100644 arch/arm64/boot/dts/marvell/armada-7040-db-A.dts
 create mode 100644 arch/arm64/boot/dts/marvell/armada-7040-db.dts
 delete mode 100644 arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
 create mode 100644 arch/arm64/boot/dts/marvell/armada-8040-db.dts

diff --git a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
index 2244c05..66ab148 100644
--- a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
+++ b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
@@ -20,7 +20,7 @@ SoC:
       when the SoC being used is the Armada 8040
 
 Boards:
-   - "marvell,armada7040-db-A", "marvell,armada7040-db"
+   - "marvell,armada7040-db-default", "marvell,armada7040-db"
       when the board being used is the Armada 7040 Development board
 
    - "marvell,armada7040-db-B", "marvell,armada7040-db"
@@ -47,7 +47,7 @@ Boards:
    - "marvell,armada7040-pcac"
       when the board being used is the Armada 7040 pcac-db emulation platform
 
-   - "marvell,armada8040-db-A", "marvell,armada8040-db"
+   - "marvell,armada8040-db-default", "marvell,armada8040-db"
       when the board being used is the Armada 8040 Development board
 
    - "marvell,armada8040-db-C", "marvell,armada8040-db"
diff --git a/Documentation/mvebu/a7k-a8k/armada7040-db-setup.txt b/Documentation/mvebu/a7k-a8k/armada7040-db-setup.txt
index 4c6374a..d04d02f 100644
--- a/Documentation/mvebu/a7k-a8k/armada7040-db-setup.txt
+++ b/Documentation/mvebu/a7k-a8k/armada7040-db-setup.txt
@@ -18,11 +18,11 @@ To switch board configuration:
 
 Board Setup
 ------------
-For A7040 A revision SoC - Please use armada-7040-db-A configuration,
+For A7040 A revision SoC - Please use armada-7040-db configuration,
 
-|		| 	1: A		|	2:B		|	3:C		|       4:D             |       4:E             |
+|		| 	1: default (A)	|	2:B		|	3:C		|       4:D             |       4:E             |
 |---------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|
-|Device tree	|armada-7040-db-A.dtb	|armada-7040-db-B.dtb	|armada-7040-db-C.dtb	|armada-7040-db-D.dtb   |armada-7040-db-E.dtb	|
+|Device tree	|armada-7040-db.dtb	|armada-7040-db-B.dtb	|armada-7040-db-C.dtb	|armada-7040-db-D.dtb   |armada-7040-db-E.dtb	|
 |CP0-SW1:1	| ON			| OFF			| ON			| ON 			| OFF                   |
 |CP0-SW1:2	| OFF			| ON			| OFF			| OFF 			| OFF                   |
 |CP0-SW1:3	| ON			| ON			| ON			| ON			| ON                    |
@@ -35,7 +35,7 @@ For A7040 A revision SoC - Please use armada-7040-db-A configuration,
 
 DPR Setting
 ------------
-|					| 	1: A				 |	2: B:                             |
+|					| 	1: default (A)			 |	2: B:                             |
 |---------------------------------------|----------------------------------------|----------------------------------------|
 | Mpp Jumper Module (SLM1548)		| Default release (ALL DPRs 1-2 position)| Default release (ALL DPRs 1-2 position)|
 | Serdess Jumper Module (SLM1547)	| DPR152,153,154,155 - 1-2 position	 | DPR152,153,154,155 - 1-2 position      |
@@ -66,7 +66,7 @@ The tables below summarizes the interface configuration of each setup
 SerDes Configuration
 ------------------------
 
-| CP0 Lane 	| 	1: A		|   2: B	| 3: C            | 4:D		  | 5:E		  |
+| CP0 Lane 	| 	1: default (A)	|   2: B	| 3: C            | 4:D		  | 5:E		  |
 |---------------|-----------------------|---------------|-----------------|---------------|---------------|
 | 0		| SGMII2		| PICE0 (x4)	| PCIE0 (x1)      |  PCIE0 (x1)   | PCIE0 (x1)	  |
 | 1		| USB3_HOST0		| PCIE0 (x4)	| USB3_HOST0      |  USB3_DEVICE0 | SATA0	  |
@@ -84,7 +84,7 @@ Multi-purpose pin configurations
 --------------------------------
 
 Setup  #	|	1	|	2	|       3       |	4	|	5	|
-Name   #	|  	A 	|	B	|       C       |	D	|	E	|
+Name   #	|  default (A) 	|	B	|       C       |	D	|	E	|
 -------------------------------------------------------------------------------------------------
 AP806:				|		|		|		|		|
 -------------------------------------------------------------------------------------------------
@@ -113,7 +113,7 @@ CP:				|		|		|		|		|
 
 Network configuration
 ---------------------
-1: A:
+1: default (A):
  Interface	| CP#	| PPv2 Port	| GMAC Port	| Board Interface		|
  ---------------|-------|---------------|---------------|-------------------------------|
  eth0		| CP0	| eth0		| 0		| SFI (10G)			|
diff --git a/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt b/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt
index d35fc63..d17b62a 100644
--- a/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt
+++ b/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt
@@ -19,16 +19,16 @@ To switch board configuration:
 Board Setup
 ------------
 
-For A8040 A1 revision SoC - Please use armada-8040-db-A configuration,
+For A8040 A1 revision SoC - Please use armada-8040-db configuration,
 			    Please use armada-8040-db-B configuration,
 			    Please use armada-8040-db-C configuration,
 			    Please use armada-8040-db-D configuration,
 			    Please use armada-8040-db-E configuration
 For A8040 A0 revision SoC - Please use armada-8040-db-A0 configuration
 
-|		| 1: A			| 3: B			| 4: C			| 5: D			| 6: E			|
+|		| 1: default (A)	| 3: B			| 4: C			| 5: D			| 6: E			|
 |---------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|
-|Device tree	|armada-8040-db-A.dtb	|armada-8040-db-B.dtb	|armada-8040-db-C.dtb	|armada-8040-db-D.dtb	|armada-8040-db-E.dtb	|
+|Device tree	|armada-8040-db.dtb	|armada-8040-db-B.dtb	|armada-8040-db-C.dtb	|armada-8040-db-D.dtb	|armada-8040-db-E.dtb	|
 |CP0-SW1:1	| OFF			| OFF			| OFF			| OFF			| OFF			|
 |CP0-SW1:2	| OFF			| OFF			| OFF			| OFF			| OFF			|
 |CP0-SW1:3	| OFF			| OFF			| OFF			| OFF			| OFF			|
@@ -48,7 +48,7 @@ For A8040 A0 revision SoC - Please use armada-8040-db-A0 configuration
 
 DPR Setting
 ------------
-|						| 1: A		| 3: B			| 4: c				|
+|						| 1: default (A)| 3: B			| 4: c				|
 |-----------------------------------------------|---------------|-----------------------|-------------------------------|
 | Jumper-module (CON44): SLM1564		| DPR all = 1-2	| DPR all = 1-2		| DPR 96-108 = 2-3		|
 | Serdes CP0 site module (CON42): SLM1521	| DPR 2-5 = 1-2	| DPR 2-5 = 2-3		| DPR 2-5 = 1-2			|
@@ -69,7 +69,7 @@ The tables below summarizes the interface configuration of each setup
 SerDes Configuration
 ------------------------
 
- CP0 Lane	| 1: A			| 3: B			| 4: C			| 5: D			| 6: E			|
+ CP0 Lane	| 1: default (A)	| 3: B			| 4: C			| 5: D			| 6: E			|
 ----------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|
  0		| PCIe0 (x1)		| PCIe0 (x4)		| PCIe0 (x1)		| PCIe0 (x1)		| PCIe0 (x1)		|
  1		| SATA0			| PCIe0 (x4)		| SATA0			| SATA0			| SATA0			|
@@ -81,7 +81,7 @@ SerDes Configuration
 - USB2_H0 (UTMI only)
 - USB2_H1 (UTMI & USB3 SerDes)
 
- CP1 Lane	| 1: A			| 3: B			| 4: C			| 5: D			| 6: E			|
+ CP1 Lane	| 1: default (A)	| 3: B			| 4: C			| 5: D			| 6: E			|
 ----------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|
  0		| PCIe0 (x1)		| PCIe0 (x1)		| PCIe0 (x1)		| PCIe0 (x1)		| PCIe0 (x1)		|
  1		| SATA0			| USB3_HOST0		| SATA0			| SATA0			| SATA0			|
@@ -96,14 +96,14 @@ SerDes Configuration
 Multi-purpose pin configurations
 --------------------------------
 
- AP806 pin	| 1: A			| 3: B			| 4: C			| 5: D			| 6: E			|
+ AP806 pin	| 1: default (A)	| 3: B			| 4: C			| 5: D			| 6: E			|
 ----------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|
     AP-SDIO	| [0-10]		| [0-10]		| [0-10]		| [0-10]		| N/C			|
     I2C0	| N/C			| N/C			| N/C			| N/C			| N/C			|
     AP-SPI0	| N/C			| N/C			| N/C			| N/C			| [0,3]			|
     AP-UART0	| [11,19]		| [11,19]		| [11,19]		| [11,19]		| [11,19]		|
 
- CP0 pin	| 1: A			| 3: B			| 4: C			| 5: D			| 6: E			|
+ CP0 pin	| 1: default (A)	| 3: B			| 4: C			| 5: D			| 6: E			|
 ----------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|
     CP0-SMI	| [32,34]		| [32,34]		| [32,34]		| [32,34]		| N/C			|
     CP0-XSMI	| [42-43]		| [42-43]		| [42-43]		| [42-43]		| [42-43]		|
@@ -122,7 +122,7 @@ GPIOS:
 	- 35: MSS_GPIO[3]: MSS_PWDN (only in configuration 1,4,5,6)
 	- 36: MSS_GPIO[5]: MSS_VTT_EN (only in configuration 1,4,5,6)
 
- CP1 pin	| 1: A			| 3: B			| 4: C			| 5: D			| 6: E			|
+ CP1 pin	| 1: default (A)	| 3: B			| 4: C			| 5: D			| 6: E			|
 ----------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|
     CP1-RGMII0	| [0-11]		| [0-11]		| N/C			| N/C			| [0-11]		|
     CP1-SMI	| [27,31]		| [27,31]		| N/C			| N/C			| [27,31]		|
@@ -136,7 +136,7 @@ GPIOS:
 
 Network configuration
 ---------------------
-1: A:
+1: default (A):
  Interface	| CPN#	| PPv2 Port	| GMAC Port	| Board Interface		|
  ---------------|-------|---------------|---------------|-------------------------------|
  eth0		| 0	| 0		| 0		| 10G (Lane 2 connected to SFP)	|
diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index c240bb3..04c9c80 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -8,7 +8,7 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-db-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-db-C.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-community.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-37x0-customer.dtb
-dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-A.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-C.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-D.dtb
@@ -18,7 +18,7 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-all-pcie.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-70x0-customer.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-pcac.dtb
 dtb-$(CONFIG_ARCH_MVEBU_PD) += armada-7040-pd.dtb
-dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-A.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-C.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-D.dtb
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-A.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-A.dts
deleted file mode 100644
index 62063f4..0000000
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-A.dts
+++ /dev/null
@@ -1,177 +0,0 @@
-/*
- * Copyright (C) 2016 Marvell Technology Group Ltd.
- *
- * This file is dual-licensed: you can use it either under the terms
- * of the GPLv2 or the X11 license, at your option. Note that this dual
- * licensing only applies to this file, and not this project as a
- * whole.
- *
- *  a) This library is free software; you can redistribute it and/or
- *     modify it under the terms of the GNU General Public License as
- *     published by the Free Software Foundation; either version 2 of the
- *     License, or (at your option) any later version.
- *
- *     This library is distributed in the hope that it will be useful,
- *     but WITHOUT ANY WARRANTY; without even the implied warranty of
- *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- *     GNU General Public License for more details.
- *
- * Or, alternatively,
- *
- *  b) Permission is hereby granted, free of charge, to any person
- *     obtaining a copy of this software and associated documentation
- *     files (the "Software"), to deal in the Software without
- *     restriction, including without limitation the rights to use,
- *     copy, modify, merge, publish, distribute, sublicense, and/or
- *     sell copies of the Software, and to permit persons to whom the
- *     Software is furnished to do so, subject to the following
- *     conditions:
- *
- *     The above copyright notice and this permission notice shall be
- *     included in all copies or substantial portions of the Software.
- *
- *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
- *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
- *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
- *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
- *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
- *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
- *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
- *     OTHER DEALINGS IN THE SOFTWARE.
- */
-
-/*
- * Device Tree file for Marvell Armada 7040 development board
- * This board file supports the A configuration of the board
- */
-
-#include "armada-7040-db.dtsi"
-
-/ {
-	model = "Marvell Armada-7040 development board A setup";
-	compatible = "marvell,armada7040-db-A", "marvell,armada7040-db", "marvell,armada7040",
-			"marvell,armada-ap806-quad", "marvell,armada-ap806";
-	#address-cells = <2>;
-	#size-cells = <2>;
-
-	chosen { };
-
-	ap806 {
-		config-space {
-			sdhci@6e0000 {
-				status = "okay";
-			};
-			serial@512000 {
-				status = "okay";
-			};
-			i2c@511000 {
-				status = "disabled";
-				clock-frequency = <100000>;
-			};
-		};
-	};
-
-	cpn-110-master {
-		config-space {
-			serial@702000 {
-				status = "disabled";
-			};
-			serial@702100 {
-				status = "disabled";
-			};
-			serial@702200 {
-				status = "disabled";
-			};
-			serial@702300 {
-				status = "disabled";
-			};
-			sata: sata@540000 {
-				status = "okay";
-
-				sata-port@0 {
-					status = "disabled";
-				};
-				sata-port@1 {
-					phys = <&comphy0 3 COMPHY_SATA1>;
-					phy-names = "comphy";
-					status = "okay";
-				};
-			};
-			usb3h0: usb3@500000 {
-				status = "okay";
-			};
-			usb3h1: usb3@510000 {
-				status = "okay";
-			};
-			u3d@520000 {
-				status = "disabled";
-			};
-			udc@524100 {
-				status = "disabled";
-			};
-			sdhci@780000 {
-				status = "okay";
-			};
-			spi@700680 {
-				status = "okay";
-			};
-			i2c1: i2c@701000 {
-				status = "okay";
-				clock-frequency = <100000>;
-			};
-			i2c2: i2c@701100 {
-				status = "disabled";
-			};
-
-			mdio@12a200 {
-				#address-cells = <1>;
-				#size-cells = <0>;
-				phy2: ethernet-phy@2 {
-					reg = <0>;
-				};
-				phy3: ethernet-phy@3 {
-					reg = <1>;
-				};
-			};
-			gop {
-				emac0: mac0 {
-					phy-mode = "sfi"; /* lane-2 */
-					phys = <&comphy0 2 COMPHY_SFI>;
-					phy-names = "comphy";
-				};
-				emac2: mac2 {
-					phy-mode = "sgmii"; /* lane-0 */
-					phy = <&phy2>;
-					phys = <&comphy0 0 COMPHY_SGMII1>;
-					phy-names = "comphy";
-				};
-				emac3: mac3 {
-					phy-mode = "rgmii"; /* rgmii-1 */
-					phy = <&phy3>;
-				};
-			};
-			ppv22@000000 {
-				l4_chksum_jumbo_port = <0>;
-				eth0: eth0@010000 {
-					status = "okay";
-				};
-				eth1: eth1@020000 {
-					status = "okay";
-				};
-				eth2: eth2@030000 {
-					status = "okay";
-				};
-			};
-			eip197: eip197@800000 {
-				status = "okay";
-			};
-			comphy {
-				status = "okay";
-			};
-		};
-
-		pcie@0x640000 {
-			status = "okay";
-		};
-	};
-};
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db.dts b/arch/arm64/boot/dts/marvell/armada-7040-db.dts
new file mode 100644
index 0000000..a93883e
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db.dts
@@ -0,0 +1,177 @@
+/*
+ * Copyright (C) 2016 Marvell Technology Group Ltd.
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPLv2 or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This library is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This library is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ * Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * Device Tree file for Marvell Armada 7040 development board
+ * This board file supports the default (A)  configuration of the board
+ */
+
+#include "armada-7040-db.dtsi"
+
+/ {
+	model = "Marvell Armada-7040 development board default (A) setup";
+	compatible = "marvell,armada7040-db-default", "marvell,armada7040-db", "marvell,armada7040",
+			"marvell,armada-ap806-quad", "marvell,armada-ap806";
+	#address-cells = <2>;
+	#size-cells = <2>;
+
+	chosen { };
+
+	ap806 {
+		config-space {
+			sdhci@6e0000 {
+				status = "okay";
+			};
+			serial@512000 {
+				status = "okay";
+			};
+			i2c@511000 {
+				status = "disabled";
+				clock-frequency = <100000>;
+			};
+		};
+	};
+
+	cpn-110-master {
+		config-space {
+			serial@702000 {
+				status = "disabled";
+			};
+			serial@702100 {
+				status = "disabled";
+			};
+			serial@702200 {
+				status = "disabled";
+			};
+			serial@702300 {
+				status = "disabled";
+			};
+			sata: sata@540000 {
+				status = "okay";
+
+				sata-port@0 {
+					status = "disabled";
+				};
+				sata-port@1 {
+					phys = <&comphy0 3 COMPHY_SATA1>;
+					phy-names = "comphy";
+					status = "okay";
+				};
+			};
+			usb3h0: usb3@500000 {
+				status = "okay";
+			};
+			usb3h1: usb3@510000 {
+				status = "okay";
+			};
+			u3d@520000 {
+				status = "disabled";
+			};
+			udc@524100 {
+				status = "disabled";
+			};
+			sdhci@780000 {
+				status = "okay";
+			};
+			spi@700680 {
+				status = "okay";
+			};
+			i2c1: i2c@701000 {
+				status = "okay";
+				clock-frequency = <100000>;
+			};
+			i2c2: i2c@701100 {
+				status = "disabled";
+			};
+
+			mdio@12a200 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				phy2: ethernet-phy@2 {
+					reg = <0>;
+				};
+				phy3: ethernet-phy@3 {
+					reg = <1>;
+				};
+			};
+			gop {
+				emac0: mac0 {
+					phy-mode = "sfi"; /* lane-2 */
+					phys = <&comphy0 2 COMPHY_SFI>;
+					phy-names = "comphy";
+				};
+				emac2: mac2 {
+					phy-mode = "sgmii"; /* lane-0 */
+					phy = <&phy2>;
+					phys = <&comphy0 0 COMPHY_SGMII1>;
+					phy-names = "comphy";
+				};
+				emac3: mac3 {
+					phy-mode = "rgmii"; /* rgmii-1 */
+					phy = <&phy3>;
+				};
+			};
+			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
+				eth0: eth0@010000 {
+					status = "okay";
+				};
+				eth1: eth1@020000 {
+					status = "okay";
+				};
+				eth2: eth2@030000 {
+					status = "okay";
+				};
+			};
+			eip197: eip197@800000 {
+				status = "okay";
+			};
+			comphy {
+				status = "okay";
+			};
+		};
+
+		pcie@0x640000 {
+			status = "okay";
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
deleted file mode 100644
index b825e2d..0000000
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
+++ /dev/null
@@ -1,238 +0,0 @@
-/*
- * Copyright (C) 2016 Marvell Technology Group Ltd.
- *
- * This file is dual-licensed: you can use it either under the terms
- * of the GPLv2 or the X11 license, at your option. Note that this dual
- * licensing only applies to this file, and not this project as a
- * whole.
- *
- *  a) This library is free software; you can redistribute it and/or
- *     modify it under the terms of the GNU General Public License as
- *     published by the Free Software Foundation; either version 2 of the
- *     License, or (at your option) any later version.
- *
- *     This library is distributed in the hope that it will be useful,
- *     but WITHOUT ANY WARRANTY; without even the implied warranty of
- *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- *     GNU General Public License for more details.
- *
- * Or, alternatively,
- *
- *  b) Permission is hereby granted, free of charge, to any person
- *     obtaining a copy of this software and associated documentation
- *     files (the "Software"), to deal in the Software without
- *     restriction, including without limitation the rights to use,
- *     copy, modify, merge, publish, distribute, sublicense, and/or
- *     sell copies of the Software, and to permit persons to whom the
- *     Software is furnished to do so, subject to the following
- *     conditions:
- *
- *     The above copyright notice and this permission notice shall be
- *     included in all copies or substantial portions of the Software.
- *
- *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
- *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
- *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
- *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
- *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
- *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
- *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
- *     OTHER DEALINGS IN THE SOFTWARE.
- */
-
-/*
- * Device Tree file for Marvell Armada 8040 development board
- * This file supports option A configuration
- * CP0_RGMII1,CP1_RGMII0,CP0_EMMC,CP1_SPI(BOOT),CP1_UART,CP0_I2C,AP_eMMC,AP_UART
- */
-
-#include "armada-8040-db.dtsi"
-
-/ {
-	model = "Marvell Armada-8040 development board A setup";
-	compatible = "marvell,armada8040-db-A", "marvell,armada8040-db", "marvell,armada8040",
-			"marvell,armada-ap806-quad", "marvell,armada-ap806";
-	#address-cells = <2>;
-	#size-cells = <2>;
-
-	chosen { };
-
-	ap806 {
-		config-space {
-			serial@512000 {
-				status = "okay";
-			};
-			sdhci@6e0000 {
-				status = "okay";
-			};
-			i2c@511000 {
-				status = "disabled";
-			};
-			spi0: spi@510600 {
-				status = "disabled";
-			};
-		};
-	};
-
-	cpn-110-master {
-		config-space {
-			sata@540000 {
-				status = "okay";
-				sata-port@0 {
-					phys = <&comphy0 1 COMPHY_SATA0>;
-					phy-names = "comphy";
-					status = "okay";
-				};
-				sata-port@1 {
-					phys = <&comphy0 3 COMPHY_SATA1>;
-					phy-names = "comphy";
-					status = "okay";
-				};
-			};
-			/* usb3h0 is USB2 only (UTMI) */
-			usb3h0: usb3@500000 {
-				status = "okay";
-			};
-			usb3h1: usb3@510000 {
-				status = "okay";
-			};
-			sdhci@780000 {
-				status = "okay";
-			};
-			i2c@701000 {
-				clock-frequency = <100000>;
-				pinctrl-names = "default";
-				pinctrl-0 = <&cpm_i2c0_pins>;
-				status = "okay";
-			};
-			mdio@12a200 {
-				#address-cells = <1>;
-				#size-cells = <0>;
-				mdio0_phy1: ethernet-phy@1 {
-					reg = <1>;
-				};
-			};
-			gop {
-				emac0: mac0 {
-					phy-mode = "sfi"; /* lane-2 */
-					phys = <&comphy0 2 COMPHY_SFI>;
-					phy-names = "comphy";
-				};
-				emac2: mac2 {
-					phy-mode = "none"; /* disabled */
-				};
-				emac3: mac3 {
-					phy-mode = "rgmii"; /* rgmii-1 */
-					phy = <&mdio0_phy1>;
-				};
-			};
-			ppv22@000000 {
-				l4_chksum_jumbo_port = <0>;
-				pinctrl-names = "default";
-				pinctrl-0 = <&cpm_ge2_rgmii_pins>;
-
-				eth0: eth0@010000 {
-					status = "okay";
-				};
-				eth1: eth1@020000 {
-					status = "disabled";
-				};
-				eth2: eth2@030000 {
-					status = "okay";
-				};
-			};
-			eip197@800000 {
-				status = "okay";
-			};
-			comphy {
-				status = "okay";
-			};
-		};
-		pcie@0x600000 {
-			status = "okay";
-		};
-		pcie@0x640000 {
-			status = "okay";
-		};
-	};
-	cpn-110-slave {
-		config-space {
-			/* usb3h0_1 is USB2 only (UTMI) */
-			usb3h0_1: usb3@500000 {
-				status = "okay";
-			};
-			serial@702000 {
-				status = "okay";
-			};
-			sata@540000 {
-				status = "okay";
-				sata-port@0 {
-					phys = <&comphy1 1 COMPHY_SATA0>;
-					phy-names = "comphy";
-					status = "okay";
-				};
-				sata-port@1 {
-					phys = <&comphy1 3 COMPHY_SATA1>;
-					phy-names = "comphy";
-					status = "okay";
-				};
-			};
-			mdio@12a200 {
-				#address-cells = <1>;
-				#size-cells = <0>;
-				mdio1_phy1: ethernet-phy@1 {
-					reg = <0>;
-				};
-			};
-			gop {
-				emac0_1: mac0 {
-					phy-mode = "sfi"; /* lane-2 */
-					phys = <&comphy1 2 COMPHY_SFI>;
-					phy-names = "comphy";
-				};
-				emac2_1: mac2 {
-					phy-mode = "rgmii"; /* rgmii-0 */
-					phy = <&mdio1_phy1>;
-				};
-				emac3_1: mac3 {
-					phy-mode = "none"; /* disabled */
-				};
-			};
-			ppv22@000000 {
-				l4_chksum_jumbo_port = <0>;
-				pinctrl-names = "default";
-				pinctrl-0 = <&cps_ge1_rgmii_pins>;
-
-				eth0_1: eth0@010000 {
-					status = "okay";
-				};
-				eth1_1: eth1@020000 {
-					status = "okay";
-				};
-				eth2_1: eth2@030000 {
-					status = "disabled";
-				};
-			};
-			eip197@800000 {
-				status = "okay";
-			};
-			comphy {
-				status = "okay";
-			};
-			spi1: spi@700680 {
-				status = "okay";
-				pinctrl-names = "default";
-				pinctrl-0 = <&cps_spi1_pins>;
-			};
-		};
-		pcie@0x600000 {
-			status = "okay";
-		};
-		pcie@0x620000 {
-			status = "okay";
-		};
-		pcie@0x640000 {
-			status = "okay";
-		};
-	};
-};
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dts b/arch/arm64/boot/dts/marvell/armada-8040-db.dts
new file mode 100644
index 0000000..78d6c66
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dts
@@ -0,0 +1,238 @@
+/*
+ * Copyright (C) 2016 Marvell Technology Group Ltd.
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPLv2 or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This library is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This library is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ * Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * Device Tree file for Marvell Armada 8040 development board
+ * This file supports option default (A)  configuration
+ * CP0_RGMII1,CP1_RGMII0,CP0_EMMC,CP1_SPI(BOOT),CP1_UART,CP0_I2C,AP_eMMC,AP_UART
+ */
+
+#include "armada-8040-db.dtsi"
+
+/ {
+	model = "Marvell Armada-8040 development board default (A) setup";
+	compatible = "marvell,armada8040-db-default", "marvell,armada8040-db", "marvell,armada8040",
+			"marvell,armada-ap806-quad", "marvell,armada-ap806";
+	#address-cells = <2>;
+	#size-cells = <2>;
+
+	chosen { };
+
+	ap806 {
+		config-space {
+			serial@512000 {
+				status = "okay";
+			};
+			sdhci@6e0000 {
+				status = "okay";
+			};
+			i2c@511000 {
+				status = "disabled";
+			};
+			spi0: spi@510600 {
+				status = "disabled";
+			};
+		};
+	};
+
+	cpn-110-master {
+		config-space {
+			sata@540000 {
+				status = "okay";
+				sata-port@0 {
+					phys = <&comphy0 1 COMPHY_SATA0>;
+					phy-names = "comphy";
+					status = "okay";
+				};
+				sata-port@1 {
+					phys = <&comphy0 3 COMPHY_SATA1>;
+					phy-names = "comphy";
+					status = "okay";
+				};
+			};
+			/* usb3h0 is USB2 only (UTMI) */
+			usb3h0: usb3@500000 {
+				status = "okay";
+			};
+			usb3h1: usb3@510000 {
+				status = "okay";
+			};
+			sdhci@780000 {
+				status = "okay";
+			};
+			i2c@701000 {
+				clock-frequency = <100000>;
+				pinctrl-names = "default";
+				pinctrl-0 = <&cpm_i2c0_pins>;
+				status = "okay";
+			};
+			mdio@12a200 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				mdio0_phy1: ethernet-phy@1 {
+					reg = <1>;
+				};
+			};
+			gop {
+				emac0: mac0 {
+					phy-mode = "sfi"; /* lane-2 */
+					phys = <&comphy0 2 COMPHY_SFI>;
+					phy-names = "comphy";
+				};
+				emac2: mac2 {
+					phy-mode = "none"; /* disabled */
+				};
+				emac3: mac3 {
+					phy-mode = "rgmii"; /* rgmii-1 */
+					phy = <&mdio0_phy1>;
+				};
+			};
+			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
+				pinctrl-names = "default";
+				pinctrl-0 = <&cpm_ge2_rgmii_pins>;
+
+				eth0: eth0@010000 {
+					status = "okay";
+				};
+				eth1: eth1@020000 {
+					status = "disabled";
+				};
+				eth2: eth2@030000 {
+					status = "okay";
+				};
+			};
+			eip197@800000 {
+				status = "okay";
+			};
+			comphy {
+				status = "okay";
+			};
+		};
+		pcie@0x600000 {
+			status = "okay";
+		};
+		pcie@0x640000 {
+			status = "okay";
+		};
+	};
+	cpn-110-slave {
+		config-space {
+			/* usb3h0_1 is USB2 only (UTMI) */
+			usb3h0_1: usb3@500000 {
+				status = "okay";
+			};
+			serial@702000 {
+				status = "okay";
+			};
+			sata@540000 {
+				status = "okay";
+				sata-port@0 {
+					phys = <&comphy1 1 COMPHY_SATA0>;
+					phy-names = "comphy";
+					status = "okay";
+				};
+				sata-port@1 {
+					phys = <&comphy1 3 COMPHY_SATA1>;
+					phy-names = "comphy";
+					status = "okay";
+				};
+			};
+			mdio@12a200 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				mdio1_phy1: ethernet-phy@1 {
+					reg = <0>;
+				};
+			};
+			gop {
+				emac0_1: mac0 {
+					phy-mode = "sfi"; /* lane-2 */
+					phys = <&comphy1 2 COMPHY_SFI>;
+					phy-names = "comphy";
+				};
+				emac2_1: mac2 {
+					phy-mode = "rgmii"; /* rgmii-0 */
+					phy = <&mdio1_phy1>;
+				};
+				emac3_1: mac3 {
+					phy-mode = "none"; /* disabled */
+				};
+			};
+			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
+				pinctrl-names = "default";
+				pinctrl-0 = <&cps_ge1_rgmii_pins>;
+
+				eth0_1: eth0@010000 {
+					status = "okay";
+				};
+				eth1_1: eth1@020000 {
+					status = "okay";
+				};
+				eth2_1: eth2@030000 {
+					status = "disabled";
+				};
+			};
+			eip197@800000 {
+				status = "okay";
+			};
+			comphy {
+				status = "okay";
+			};
+			spi1: spi@700680 {
+				status = "okay";
+				pinctrl-names = "default";
+				pinctrl-0 = <&cps_spi1_pins>;
+			};
+		};
+		pcie@0x600000 {
+			status = "okay";
+		};
+		pcie@0x620000 {
+			status = "okay";
+		};
+		pcie@0x640000 {
+			status = "okay";
+		};
+	};
+};
-- 
2.7.4

